en/guides/codablerouting.md

```diff
@@ -15,172 +15,263 @@ redirect_from:
 
-Here the store handler accepts a todo item of type `ToDo` and responds using an asynchronous completion handler. The handler then stores the received todo item in the todoStore and returns with the created todo item in the case of success, and an `Error` in the the case of failure.
+`kitura init` also creates an Xcode project for our server so we can open that now:  
+`open CodableRouting.xcodeproj`
 
-The input parameter and the response parameter in completion handler can be any Swift type you want, as long as it conforms to Codable. For example, we could use the following as our `ToDo` type:
+Now we're ready to start adding our Codable routes!
 
+### Creating the models
+Codable routing works directly with Swift value types. So we need to contruct a value type to use. A good example of this is a struct that conforms to Codable: 
 ```swift
-public struct ToDo : Codable {
-    public var title: String        // title of the todo item
-    public var order: Int           // order to display the todo item
-    public var completed: Bool      // has the item been completed
+struct User: Codable {
+    let name: String
+    let age: Int
 }
 ```
 
-At this point the server has stored the item, but the client has no way of referring to that specific item on the server if it wants to get, update, or delete that item. In order to do that, some kind of unique identifier is required.
+It's good practice to keep value types (in this case models) and logic separated, we will create a new file called `Models.swift` as follows:  
+1. In Xcode right click on the `Application` folder and select `New File`
+2. Select `Swift File` in the UI pop up and click next
+3. In the `Save As` field, name the file `Models.swift`
+4. Ensure Group is `Application` and target is `Application` as well
+5. Click `Create` to create the file
 
-Returning an identifier to the client for the stored data is typically done in one of two ways:
-
-* ### Adding an additional Optional field to the type
-Here the ToDo struct would be updated to add an extra field:
+For now we're going to create a very simple model: 
+1. Open `Models.swift`
+2. Copy in the following code - 
 ```swift
-public struct ToDo : Codable {
-    public var title: String       
-    public var order: Int      
-    public var completed: Bool
-    public var id: Int?         // Additional field for identifier
+struct User: Codable {
+    let name: String
+    let age: Int
 }
 ```
 
-And the storeHandler function would then set it:
+This provides a simple representation of a user that has two properties, a name and an age. In real applications models tend to be more complex, and typically would be made up of several models.  
+
+Now we have a model we can use, next we can create our Codable routes!
```

Comment:
I think you already said "Now you're ready to create Codable routes" on line 47 (or similar)?